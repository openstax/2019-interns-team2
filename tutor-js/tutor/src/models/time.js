import { computed, decorate } from 'mobx';
import { now as getNow } from 'mobx-utils';

let shiftMs = 0;
const defaultResolution = 1000 * 60; // one minute resolution
let timeResolution = defaultResolution;

export function setNow(now) {
  shiftMs = now.getTime() - (new Date()).getTime();
}

export function setResolution(r = defaultResolution) {
  timeResolution = r;
}

const Store = {

  get now() {
    const now = getNow(timeResolution);
    return new Date(now + shiftMs);
  },

};

decorate(Store, {
  now: computed,
});

export default Store;
