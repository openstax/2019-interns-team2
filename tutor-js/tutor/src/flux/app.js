// Store for api requests and responses
import _ from 'underscore';

import { makeSimpleStore } from './helpers';


const AppConfig = {
  resetServerErrors() {
    return delete this._currentServerError;
  },

  setServerError(errorResponse) {
    const { status, data } = errorResponse;
    this._currentServerError = errorResponse;
    return this.emit('server-error', status, data);
  },

  exports: {
    getError() { return this._currentServerError; },

    errorNavigation() {
      if (!this._currentServerError) { return {}; }
      const { status, request } = this._currentServerError;
      if (status === 403) {
        return { href: '/' };
      } else {
        const isGET404 = (status === 404) && (request.method === 'GET');
        const isInRange = 400 <= status && status < 600;
        return { shouldReload: false };
      }
    },
  },
};

const { actions, store } = makeSimpleStore(AppConfig);
export { actions as AppActions, store as AppStore };
