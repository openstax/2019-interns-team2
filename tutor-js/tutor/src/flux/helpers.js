import _ from 'underscore';
import flux from 'flux-react';

const LOADING = 'loading';
const LOADED  = 'loaded';
const FAILED  = 'failed';
const SAVING  = 'saving';
const DELETING = 'deleting';
const DELETED = 'deleted';

const STATES = {
  LOADING,
  LOADED,
  FAILED,
  SAVING,
  DELETING,
  DELETED,
};

let idCounter = 0;
const CREATE_KEY = () => `_CREATING_${idCounter++}`;

const isNew = id => /_CREATING_/.test(id);

const CrudConfig = function() {
  return {
    _asyncStatus: {},
    _local: {},
    _changed: {},
    _errors: {},
    _reload: {},

    // If the specific type needs to do something else to the object:
    // _loaded : (obj, id) ->
    // _saved : (obj, id) ->
    // _reset : ->

    reset() {
      this._asyncStatus = {};
      this._local = {};
      this._changed = {};
      this._errors = {};
      this._reload = {};

      if (typeof this._reset === 'function') {
        this._reset();
      }
      return this.emitChange();
    },

    dontReload(id) {
      return (this._asyncStatus[id] === LOADED) && this._HACK_DO_NOT_RELOAD;
    },

    FAILED(status, msg, id, ...args) {
      this._asyncStatus[id] = FAILED;
      this._errors[id] = msg;
      if (typeof this._failed === 'function') {
        this._failed({ status, msg }, id, ...Array.from(args));
      }

      if (status !== 0) { // indicates network failure
        delete this._local[id];
        return this.emitChange();
      }
    },

    load(id) {
      return this._load(id);
    },

    // allow aliasing load without triggering api call.
    _load(id) {
      // Add a shortcut for unit testing
      if (this.dontReload(id)) { return; }
      this._reload[id] = false;
      this._asyncStatus[id] = LOADING;
      return this.emitChange();
    },

    loaded(obj, id, ...args) {
      // id = obj.id
      this._asyncStatus[id] = LOADED;

      if (obj) {
        // If the specific type needs to do something else to the object:
        this._local[id] = (typeof this._loaded === 'function' ? this._loaded(obj, id, ...Array.from(args)) : undefined) || obj;
      }

      return this.emitChange();
    },

    save(id, obj) {
      // Note: id could be isNew()
      this._asyncStatus[id] = SAVING;
      return this.emitChange();
    },

    saved(result, id, ...args) {
      // id = result.id
      this._asyncStatus[id] = LOADED; // TODO: Maybe make this SAVED

      // If the specific type needs to do something else to the object:
      const obj = typeof this._saved === 'function' ? this._saved(result, id, ...Array.from(args)) : undefined;
      if (obj) { result = obj; }

      if (result) {
        this._local[id] = result;
        this._local[result.id] = result;
        delete this._changed[result.id];
      } else {
        console.warn('API WARN: Server did not return JSON after saving. Patching locally');
        // Merge all the local changes into the new local object
        this._local[id] = _.extend(this._local[id], this._changed[id]);
      }
      delete this._changed[id];
      delete this._errors[id];
      // If the specific type needs to do something else to the object:
      return this.emitChange();
    },

    create(localId, attributes = {}) {
      if (!isNew(localId)) { throw new Error('BUG: MUST provide a local id'); }
      this._local[localId] = {};
      this._changed[localId] = attributes;
      return this._asyncStatus[localId] = LOADED;
    },

    created(result, localId, ...args) {
      this._local[localId] = result; // HACK: So react component can still manipulate the same object
      this._local[result.id] = result;
      this._asyncStatus[localId] = LOADED;
      this._asyncStatus[result.id] = LOADED;
      this.emitChange();
      return (typeof this._created === 'function' ? this._created(result, localId, ...Array.from(args)) : undefined);
    },

    _change(id, obj) {
      if (this._changed[id] == null) { this._changed[id] = {}; }
      _.extend(this._changed[id], obj);
      return this.emitChange();
    },

    _save(id) {
      return this._asyncStatus[id] = SAVING;
    },

    delete(id) {
      this._asyncStatus[id] = DELETING;
      return (typeof this._delete === 'function' ? this._delete(id) : undefined);
    },

    deleted(result, id, ...args) {
      this._asyncStatus[id] = DELETED;
      delete this._local[id];
      this.emitChange();
      return (typeof this._deleted === 'function' ? this._deleted(result, id, ...Array.from(args)) : undefined);
    },

    clearChanged(id) {
      return delete this._changed[id];
    },

    HACK_DO_NOT_RELOAD(bool) { return this._HACK_DO_NOT_RELOAD = bool; },

    // Keep this here so other exports method have access to it
    _get(id) {
      const val = this._local[id];
      if (!val && (this._asyncStatus[id] !== SAVING)) { return null; }
      // Scores stores an Array unlike most other stores
      if (_.isArray(val)) {
        return val;
      } else {
        return _.extend({}, val, this._changed[id]);
      }
    },

    exports: {
      isUnknown(id) { return !this._asyncStatus[id]; },
      isLoading(id) { return this._asyncStatus[id] === LOADING; },
      isLoaded(id) { return this._asyncStatus[id] === LOADED; },
      isDeleting(id) { return this._asyncStatus[id] === DELETING; },
      isSaving(id) { return this._asyncStatus[id] === SAVING; },
      isFailed(id) { return this._asyncStatus[id] === FAILED; },
      getAsyncStatus(id) { return this._asyncStatus[id]; },
      get(id) { return this._get(id); },
      isChanged(id) { return !_.isEmpty(this._changed[id]); },
      getChanged(id) { return this._changed[id] || {}; },
      freshLocalId() { return CREATE_KEY(); },
      isNew(id) { return isNew(id); },
      reload(id) { return this._reload[id]; },
      getError(id) { return this._errors[id]; },
    },
  };
};

// Helper for creating a simple store for actions
const makeSimpleStore = function(storeConfig) {

  let actionsConfig = _.omit(storeConfig, (value, key) => (typeof value !== 'function') || (key === 'exports'));

  actionsConfig = _.keys(actionsConfig);
  const actions = flux.createActions(actionsConfig);
  storeConfig.actions = _.values(actions);
  const store = flux.createStore(storeConfig);
  return { actions, store };
};


const extendConfig = function(newConfig, origConfig) {
  if (newConfig.exports == null) { newConfig.exports = {}; }
  _.defaults(newConfig, origConfig);
  _.defaults(newConfig.exports, origConfig.exports);
  return newConfig;
};


const makeStandardStore = function(name, storeConfig) {
  extendConfig(storeConfig, new CrudConfig());
  const { actions, store } = makeSimpleStore(storeConfig);
  return { [`${name}Actions`]: actions, [`${name}Store`]: store };
};


export { CrudConfig, makeSimpleStore, extendConfig, makeStandardStore, isNew, STATES };
