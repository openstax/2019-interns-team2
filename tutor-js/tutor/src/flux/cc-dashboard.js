import _ from 'underscore';
import { CrudConfig, makeSimpleStore, extendConfig } from './helpers';
import PeriodHelper from '../helpers/period';

const DEFAULT_COURSE_TIMEZONE = 'US/Central';

const DashboardConfig = {
  exports: {
    isBlank(courseId) {
      return !_.any(__guard__(__guard__(this._get(courseId), x1 => x1.course), x => x.periods));
    },

    getPeriods(courseId) { return PeriodHelper.sort(__guard__(__guard__(this._get(courseId), x1 => x1.course), x => x.periods)); },

    chaptersForDisplay(courseId, periodId) {
      const period = _.findWhere( __guard__(__guard__(this._get(courseId), x1 => x1.course), x => x.periods), { id: periodId });
      if (!period) { return []; }
      for (let chapter of period.chapters) {
        for (let page of chapter.pages) {
          const total = page.completed + page.in_progress + page.not_started;
          page.completed_percentage = page.completed / total;
        }
        chapter.valid_sections = _.sortBy(chapter.pages, page => page.chapter_section[1] || 0).reverse();
        chapter;
      }

      return _.select(period.chapters.reverse(), chapter => chapter.valid_sections != null ? chapter.valid_sections.length : undefined);
    },
  },
};


extendConfig(DashboardConfig, new CrudConfig());
const { actions, store } = makeSimpleStore(DashboardConfig);
export { actions as CCDashboardActions, store as CCDashboardStore };

function __guard__(value, transform) {
  return (typeof value !== 'undefined' && value !== null) ? transform(value) : undefined;
}
