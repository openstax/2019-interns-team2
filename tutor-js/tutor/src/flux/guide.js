import { CrudConfig, makeSimpleStore, extendConfig } from './helpers';

const CourseGuideConfig = {};

extendConfig(CourseGuideConfig, new CrudConfig());
const { actions, store } = makeSimpleStore(CourseGuideConfig);
export { actions as CourseGuideActions, store as CourseGuideStore };
