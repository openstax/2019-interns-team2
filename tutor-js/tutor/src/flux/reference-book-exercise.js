import { CrudConfig, makeSimpleStore, extendConfig } from './helpers';
import _ from 'underscore';

const LOADING = 'loading';
const QUERY_START_STRING = '?q=';

const combineQueries = function(multipleUrls) {
  const params = {};
  const tags = _.map(multipleUrls, function(url) {
    const queryString = decodeURIComponent(url.split(QUERY_START_STRING)[1]);
    let [param, value] = Array.from(queryString.split(':'));
    // trim individual tag of double quotes
    value = value.replace(/^"(.+(?="$))"$/, '$1');
    if (params[param] == null) { params[param] = []; }
    params[param].push(value);

    if (param === 'tag') { return value; }
  });
  const urlsAndTags = _.object(multipleUrls, tags);
  const queryStrings = _.map(params, (values, paramKey) =>
    // join values and wrap all values with a single pair of double quotes
    `${paramKey}:\"${values.join(',')}\"`
  );
  const queryString = queryStrings.join(' ');

  return { queryString, urlsAndTags };
};

const getMultipleUrl = function(multipleUrls, baseUrl) {
  const { queryString } = combineQueries(multipleUrls);

  return `${baseUrl}?q=${queryString}`;
};

const ReferenceBookExerciseConfig = {
  _toSeparate: {},

  loadMultiple(multipleUrls) {
    const baseUrl = _.first(multipleUrls).split(QUERY_START_STRING)[0];
    const { queryString, urlsAndTags } = combineQueries(multipleUrls, baseUrl);
    const url = `${baseUrl}?q=${queryString}`;

    return this._toSeparate[url] = urlsAndTags;
  },

  _loaded(obj, id) {
    if (this._toSeparate[id] != null) {
      _.each(this._toSeparate[id], (tag, url) => {
        const exercise = _.find(obj.items, item => item.tags.indexOf(tag) > -1);
        const exerciseObj =
          { items: [exercise] };
        return this.loaded(exerciseObj, url);
      });

      delete this._toSeparate[id];
      this.emit('loaded.multiple');
    } else {
      this.emit(`loaded.${id}`);
    }

    return obj;
  },

  exports: {
    isQueued(id) {
      return _.chain(this._toSeparate)
        .find(urlsAndTags => urlsAndTags[id] != null)
        .isObject()
        .value();
    },
  },
};

extendConfig(ReferenceBookExerciseConfig, new CrudConfig());
const { actions, store } = makeSimpleStore(ReferenceBookExerciseConfig);
export { actions as ReferenceBookExerciseActions, store as ReferenceBookExerciseStore };
