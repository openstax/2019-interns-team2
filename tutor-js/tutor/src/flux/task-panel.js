import _ from 'underscore';
import { makeSimpleStore } from './helpers';

import { TaskStore } from '../flux/task';
import { StepTitleStore } from '../flux/step-title';
import { TaskHelper } from 'shared';

const TaskPanel = {
  _steps: {},

  loaded(task, taskId) {
    this._steps[taskId] = TaskHelper.mapSteps(task);
    return this.emit('loaded');
  },

  sync(taskId) {
    const task = TaskStore.get(taskId);
    task.steps = TaskStore.getSteps(taskId);
    return this.loaded(task, taskId);
  },

  _get(taskId) {
    return _.where(this._steps[taskId], { isAvailable: true });
  },

  exports: {
    get(id) { return this._get(id); },
    getStep(id, stepIndex) { return this._get(id)[stepIndex]; },
    getStepByKey(id, stepKey) { return this.exports.getStep.call(this, id, stepKey - 1); },

    getNextStepTitle(id, taskStepId) {
      const steps = this._get(id);
      const index = _.findIndex(steps, { id: taskStepId });
      const nextStep = index === -1 ? null : steps[index + 1];
      if (nextStep) { return StepTitleStore.getTitleForCrumb(nextStep); } else { return ''; }
    },

    getStepIndex(id, stepInfo) {
      const steps = this._get(id);
      const stepIndex = _.findIndex(steps, stepInfo);
      if (stepIndex === -1) { return null; }
      return stepIndex;
    },

    getStepKey(id, stepInfo) {
      return this.exports.getStepIndex.call(this, id, stepInfo) + 1;
    },

    getProgress(id, stepIndex) {
      const steps = this._steps[id];

      if (steps.length) {
        return ((stepIndex + 1) / (steps.length)) * 100;
      } else {
        return 0;
      }
    },

    getTitlesForStepIndex(taskId, stepIndex) {
      let previous;
      const crumbs = this._get(taskId); // TaskPanelStore.get(@props.id)
      if (stepIndex >= 0) { previous = crumbs[stepIndex - 1]; }
      const next = crumbs[stepIndex + 1];

      return {
        previous: previous && StepTitleStore.getTitleForCrumb(previous),
        current: StepTitleStore.getTitleForCrumb(crumbs[stepIndex]),
        next: next && StepTitleStore.getTitleForCrumb(next),
      };
    },
  },
};

const { actions, store } = makeSimpleStore(TaskPanel);

TaskStore.on('step.completed', (stepId, taskId) => actions.sync(taskId));

export { actions as TaskPanelActions, store as TaskPanelStore };
