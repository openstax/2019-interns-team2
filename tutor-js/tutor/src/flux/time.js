// Store for getting the current server time.
// Defaults to browser time if a server request has not been made yet.

import { setNow } from '../models/time';

import flux from 'flux-react';

import { makeSimpleStore } from './helpers';

const TutorDateFormat = 'MM/DD/YYYY';

const TimeConfig = {
  reset() {
    return this._shiftMs = 0;
  },

  setNow(now, localNow = new Date()) {
    setNow(now, localNow); // also notify the mobx store
    // # called by API
    return this._shiftMs = now.getTime() - localNow.getTime();
  },

  setFromString(txtDate, localNow = new Date()) {
    if (!txtDate) { return; }
    const date = new Date(txtDate);
    if (isNaN(date.getTime())) { // "Invalid Date"
      return __guardMethod__(console, 'warn', o => o.warn(`Attempted to set invalid date ${txtDate} on TimeStore`));
    } else {
      return this.setNow(date, localNow);
    }
  },

  exports: {
    getNow(localNow = new Date()) {
      const shift = this._shiftMs || 0;
      return new Date(localNow.getTime() + shift);
    },

    getFormat() { return TutorDateFormat; },
  },
};

const { actions, store } = makeSimpleStore(TimeConfig);
export { actions as TimeActions, store as TimeStore };

function __guardMethod__(obj, methodName, transform) {
  if (typeof obj !== 'undefined' && obj !== null && typeof obj[methodName] === 'function') {
    return transform(obj, methodName);
  } else {
    return undefined;
  }
}