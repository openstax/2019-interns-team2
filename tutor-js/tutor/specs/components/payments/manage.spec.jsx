import { C } from '../../helpers';
import moment from 'moment';
import Payments from '../../../src/components/payments/manage';
import Purchases from '../../../src/models/purchases';
import Router from '../../../src/helpers/router';
import PaymentModel from '../../../src/models/payments';
import mockData from '../../../api/purchases.json';
import { TimeStore } from '../../../src/flux/time';
import TimeHelper from '../../../src/helpers/time';
import chronokinesis from 'chronokinesis';

jest.mock('../../../src/helpers/router');
jest.mock('../../../src/models/payments');


describe('Student Payments Management', () => {
  beforeEach(() => {
    PaymentModel.config = {
      is_enabled: true,
      base_url: 'url-for-test',
    };
    chronokinesis.travel(new Date('2017-07-01'));

    Router.currentParams.mockReturnValue({ courseId: '2' });
    Router.makePathname.mockImplementation(() => '/foo');
    Purchases.onLoaded({ data: mockData });
  });

  afterEach(() => {
    chronokinesis.reset();
  });

  it('renders and matches snapshot', () => {
    expect.snapshot(<C noRef><Payments /></C>).toMatchSnapshot();
  });

  it('renders empty when no payments', () => {
    Purchases.clear();
    expect.snapshot(<C noRef><Payments /></C>).toMatchSnapshot();
  });

});
