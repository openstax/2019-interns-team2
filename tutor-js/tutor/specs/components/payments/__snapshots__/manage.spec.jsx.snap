// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Payments Management renders and matches snapshot 1`] = `
<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="manage-payments container"
      >
        <header>
          <h1>
            Manage payments
          </h1>
          <a
            className="btn btn-default"
            href="/foo"
            onClick={[Function]}
          >
            Back to Dashboard
          </a>
        </header>
        <table
          className="table table-striped"
        >
          <thead>
            <tr>
              <th>
                Item
              </th>
              <th>
                Transaction date
              </th>
              <th>
                Order number
              </th>
              <th
                className="right"
              >
                Amount
              </th>
              <td />
            </tr>
          </thead>
          <tbody>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                July 13th 2017
              </td>
              <td>
                33uqbxbi324n
              </td>
              <td
                className="right total"
              >
                13.41
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="33uqbxbi324n"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="33uqbxbi324n"
                  href="url-for-test/invoice/33uqbxbi324n"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                July 13th 2017
              </td>
              <td>
                nuvwqun23ztq
              </td>
              <td
                className="right total"
              >
                13.41
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="nuvwqun23ztq"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="nuvwqun23ztq"
                  href="url-for-test/invoice/nuvwqun23ztq"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                July 13th 2017
              </td>
              <td>
                ru6ifviicwki
              </td>
              <td
                className="right total"
              >
                13.41
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="ru6ifviicwki"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="ru6ifviicwki"
                  href="url-for-test/invoice/ru6ifviicwki"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                OpenStax Tutor
              </td>
              <td>
                July 14th 2017
              </td>
              <td>
                fxa4s2kj75fr
              </td>
              <td
                className="right total"
              >
                13.41
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="fxa4s2kj75fr"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="fxa4s2kj75fr"
                  href="url-for-test/invoice/fxa4s2kj75fr"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                July 16th 2017
              </td>
              <td>
                3dmy8nu4v4bg
              </td>
              <td
                className="right total"
              >
                357.25
              </td>
              <td
                className="refund"
              >
                <button
                  className="async-button btn btn-primary"
                  data-identifier="3dmy8nu4v4bg"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Request Refund
                </button>
              </td>
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="3dmy8nu4v4bg"
                  href="url-for-test/invoice/3dmy8nu4v4bg"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                July 16th 2017
              </td>
              <td>
                kjxdvs9wd7rw
              </td>
              <td
                className="right total"
              >
                1388.07
              </td>
              <td />
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="kjxdvs9wd7rw"
                  href="url-for-test/invoice/kjxdvs9wd7rw"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className="refunded"
            >
              <td>
                unit test product refund
              </td>
              <td>
                July 18th 2017
              </td>
              <td>
                kjxdvs9wd7rw:refund
              </td>
              <td
                className="right total"
              >
                -1388.07
              </td>
              <td />
              <td />
            </tr>
            <tr
              className=""
            >
              <td>
                unit test product
              </td>
              <td>
                July 16th 2017
              </td>
              <td>
                cufu6hr4xicy
              </td>
              <td
                className="right total"
              >
                1591.48
              </td>
              <td />
              <td>
                <a
                  className="btn btn-link"
                  data-identifier="cufu6hr4xicy"
                  href="url-for-test/invoice/cufu6hr4xicy"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                >
                  Invoice
                </a>
              </td>
            </tr>
            <tr
              className="refunded"
            >
              <td>
                unit test product refund
              </td>
              <td>
                July 19th 2017
              </td>
              <td>
                cufu6hr4xicy:refund
              </td>
              <td
                className="right total"
              >
                -1591.48
              </td>
              <td />
              <td />
            </tr>
          </tbody>
        </table>
        <div
          className="footer"
        >
          <a
            className="refund-policy external-icon"
            href="https://openstax.secure.force.com/help/articles/FAQ/OpenStax-Tutor-Beta-Student-Refund-Policy"
            tabIndex={0}
            target="_blank"
          >
            Refund policy for OpenStax Tutor Beta courses
          </a>
          <div
            className="help"
          >
            Need help? 
            <a
              href="mailto:support@openstax.org"
            >
              Contact Support
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    π
  </a>
</div>
`;

exports[`Student Payments Management renders empty when no payments 1`] = `
<div
  className="openstax-debug-content"
>
  <div
    data-purpose="tour-conductor-wrapper"
  >
    <div>
      <div
        className="manage-payments container"
      >
        <header>
          <h1>
            Manage payments
          </h1>
          <a
            className="btn btn-default"
            href="/foo"
            onClick={[Function]}
          >
            Back to Dashboard
          </a>
        </header>
        <div
          className="empty"
        >
          <h3>
            No payments were found for your account.
          </h3>
        </div>
        <div
          className="footer"
        >
          <a
            className="refund-policy external-icon"
            href="https://openstax.secure.force.com/help/articles/FAQ/OpenStax-Tutor-Beta-Student-Refund-Policy"
            tabIndex={0}
            target="_blank"
          >
            Refund policy for OpenStax Tutor Beta courses
          </a>
          <div
            className="help"
          >
            Need help? 
            <a
              href="mailto:support@openstax.org"
            >
              Contact Support
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="visible-when-debugging"
    >
      <div
        className="tour-spy-info"
      >
        available regions: []; region tour ids: []; audience tags: []; tour tags: []; eligible tours: []; TOUR RIDE: &lt;none&gt;
      </div>
    </div>
  </div>
  <a
    aria-hidden="true"
    className="debug-toggle-link"
    href="#spy"
    onClick={[Function]}
    tabIndex={-1}
  >
    π
  </a>
</div>
`;
